functions/client/fn_camRotate.sqf

```diff
@@ -1,36 +1,53 @@
-params ["_angleStart", "_angleEnd", "_target", "_duration", "_zoom1", "_zoom2", "_angle", "_radius", "_cclockwise"];
-
-private _timeBegin = time;
-
-private _targetPos = if (_target isEqualType []) then {_target} else {getPos _target};
-private _coords = _targetPos getPos [_radius,_angle];
-
-private _angleDelta = _angleEnd - _angleStart;
-private _angleDistance = _duration / _angleDelta * 360 * 0.1;
-
-// diag_log ["rotation: %1 + %2 + %3 + %4 + %5 + %6", _angleStart, _angleEnd, _angle, _radius, _angleDelta, _angleDistance];
-// ["rotation: %1 + %2 + %3 + %4",150,100,89,scalar NaN]
-
-
-private _camera = "camera" camCreate _coords;
-showCinemaBorder true;
-_camera cameraEffect ["internal", "BACK"];
-_camera camCommand "inertia off";
-_camera camPrepareTarget _target;
-_camera camPrepareFOV _zoom1;
-_camera camCommitPrepared 0;
-
-while {(time - _timeBegin) < _duration} do {
- 	private _targetPos = if (_target isEqualType []) then {_target} else {getPos _target};
-
-    private _coords = _targetPos getPos [_radius,_angle];
-
-    _camera camPreparePos _coords;
-    _camera camCommitPrepared 0.1;
-
-    waitUntil {camCommitted _camera};
-
-    _angle = if (_cclockwise) then {_angle - _angleDistance} else {_angle + _angleDistance};
-
-    systemChat format ["angle: %1", _angle];
-};
\ No newline at end of file
+params ["_cam", "_target", "_startingAngle", "_endAngle", "_duration", "_radius", ["_clockwise", true], ["_rise", 0]];
+
+GRAD_introCam_camRotateFinish = false;
+private _angleDistance = (_endAngle - _startingAngle) mod 360;
+private _steps = if (_clockwise) then {
+    (_angleDistance / _duration)
+}else{
+    (0-(_angleDistance / _duration))
+};
+
+private _riseSteps = if (_rise != 0) then {
+        (_rise / _duration)
+};
+
+_steps = _steps* 0.01;
+_riseSteps = _riseSteps* 0.01;
+_duration = _duration* 0.01;
+
+GRAD_introCam_camAngle = _startingAngle;
+private _pos = getPos _cam;
+private _camAttachObj = "HeliHEmpty" createVehicleLocal _pos;
+_camAttachObj setPos _pos;
+_cam attachTo [_camAttachObj, [0, 0, 0]];
+
+[
+    {
+        params ["_args", "_handle"];
+        _args params ["_camAttachObj", "_cam", "_target", "_steps", "_endAngle", "_duration", "_riseSteps", "_radius"];
+        if (GRAD_introCam_camAngle == _endAngle || {time > _duration}) exitWith {
```

Comment:
`_duration` is a misnomer, it should be called `_endTime` or something else indicating a *point* in time.